pkg/api/error.go

```diff
@@ -98,7 +99,19 @@ func handleRekorAPIError(params interface{}, code int, err error, message string
 		case http.StatusBadRequest:
 			return entries.NewCreateLogEntryBadRequest().WithPayload(errorMsg(message, code))
 		case http.StatusConflict:
-			return entries.NewCreateLogEntryConflict().WithPayload(errorMsg(message, code))
+			resp := entries.NewCreateLogEntryConflict().WithPayload(errorMsg(message, code))
+			locationFound := false
+			for _, field := range fields {
+				if locationFound {
+					existingURL := field.(strfmt.URI)
+					resp.SetLocation(existingURL)
+					break
+				} else if field.(string) == "entryURL" {
```

Comment:
Does this require the fields to be in the right order? Would it be safer to do two passes? one for the url then another for the URI?

Reply:
reviewed with @dlorenc